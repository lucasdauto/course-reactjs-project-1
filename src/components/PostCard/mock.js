export const postCardPropsMock = {
  title: "title 1",
  body: "Body 1",
  id: 1,
  cover: "img/test.png",
};
