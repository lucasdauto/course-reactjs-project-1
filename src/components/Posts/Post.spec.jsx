import { render, screen } from "@testing-library/react";
import { Posts } from '.';

import { props } from "./mock";

describe("<Post />", () => {
  it("should render posts", () => {
    render(<Posts {...props}/>);

    expect(screen.getAllByRole('heading',{ name: /title/i }))
      .toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /title/i }))
      .toHaveLength(3);
    expect(screen.getAllByText(/body/i))
      .toHaveLength(3);
    expect(screen.getByRole('img', { name: /title 3/i }))  
      .toHaveAttribute('src', 'img/img3.png');
  });

  it("should match snapshot", () => {
    const  { container } = render(<Posts {...props} />);

    expect(container.firstChield).
      toMatchSnapshot();
  });
});