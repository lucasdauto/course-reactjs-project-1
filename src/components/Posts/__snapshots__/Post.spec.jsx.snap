// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snapshot 1`] = `undefined`;
